build_runner/lib/src/asset_graph/graph.dart

```diff
@@ -309,6 +292,24 @@ class AssetGraph {
         _addOutputsForSources(buildPhases, newIds, rootPackage)
           ..addAll(transitiveRemovedIds);
 
+    // Transitively invalidates all assets.
```

Comment:
Optionally add a comment here indicating that this must be done after updates to the graph are completed.